Extensions/DataSources/src/DataSourcesHub/action-creators/DataSourcesActionCreators.tsx

```diff
@@ -0,0 +1,195 @@
+import { getClient } from 'azure-devops-extension-api/Common';
+import { DataSourceInfo,PayloadForExecuteServiceEndpointRequest,PayloadForUpdateDataSource, PayloadForUpdateDataSourceParameters, PayloadForSelectDataSource, ParamInfo } from '../states/DataSourcesExtensionState';
+
+export class DataSourcesActionCreators extends Reflux.Component{
+
+    public updateDataSourceParameters(name:string,value:string) {
+        var payload:PayloadForUpdateDataSourceParameters = {
+            name:name,
+            value:value,
+            parseError:null,
+            executeError:null,
+            result:null,
+        };       
+        DataSourcesActions.UpdateDataSourceParameters(payload)
+    }
+
+    public selectDataSource(itemtext: string, datasourceInfo: string) {
+        var Parameters:ParamInfo=this.extractParameters(datasourceInfo,null)
+        var payload:PayloadForSelectDataSource = {
+            selectedDataSource: itemtext,
+            currentInputParam: Parameters,
+            result: null,
+            parseError: null,
+            displayInfo: datasourceInfo,
+            executeError: null
+        };
+        DataSourcesActions.SelectDataSource(payload);
+    }
+
+    public updateDataSource(newValue: string,currentInputParam:ParamInfo | null) {
+        var Parameters:ParamInfo=this.extractParameters(newValue,currentInputParam)
+        var payload: PayloadForUpdateDataSource = {
+                currentInputParam: Parameters,
+                result: null,
+                parseError: null,
+                executeError: null,
+                displayInfo: newValue
+        };
+        DataSourcesActions.UpdateDataSource(payload);
+    }
+
+    public executeServiceEndpointRequest(DataSourceInfo: string|null, currentInputParam: ParamInfo|null, endpointDetails: ServiceEndpointDetails|null) {
+        var endpointId = this.getEndpointId();
+        var project = this.getProject();
+        var serviceEndpointRestClient = getClient(ServiceEndpointRestClient);
+        var dataSourceDetails: DataSourceDetails|null=null;
+        var resultTransformationDetails: ResultTransformationDetails | null =null;
+        var serviceEndpointDetails: ServiceEndpointDetails | null = null;
+        try{
+            if(DataSourceInfo!=null && currentInputParam!=null && endpointDetails!=null){
+                dataSourceDetails= this.constructDataSourceDetails(DataSourceInfo,currentInputParam);
+                resultTransformationDetails= this.constructResultTransformationDetails(DataSourceInfo);  
+                serviceEndpointDetails= endpointDetails;  
+            } 
+        }
+        catch(error){
+            var payload: PayloadForExecuteServiceEndpointRequest = {
```

Comment:
we should have seperate action for handling errors.
Store should listen to them seperately and UI should act on them seperately